### PR TITLE
GRW-1743 /Story for Car Price tier accordion

### PR DESCRIPTION
**Pull request by KarnellSchultz**
## Describe your changes

Created a Price tier component and a story for it. 

Decided not to add the light grey text you'll see in the designs as it did not have an acceptable contrast ratio.

---
**Coming later** ---> storyblok component

https://user-images.githubusercontent.com/50870173/202681276-1f10b5a2-5f7e-442d-aefd-6b4be6bd7cdb.mov

[Designs](https://www.figma.com/file/8SfzuQEJ4LJ6uKL0l7WKoE/Hedvig.com-%C2%B7-Final-Design?node-id=1289%3A5130&t=bhuRzFExMDbssqDC-0)

Screenshot 2022-11-18 at 11 30 28

## Justify why they are needed

Users want to select the tier of car insurance. 

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

[GRW-1743]: https://hedvig.atlassian.net/browse/GRW-1743?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by amandaflod — APPROVED** [review]
This looks great!

File: apps/store/src/components/CarTierSelector/CarTierSelector.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
+import { ComponentMeta, ComponentStory } from '@storybook/react'
+import { useState } from 'react'
+import * as Accordion from './CarTierSelector'
```
Comment (gustaveen) [inline apps/store/src/components/CarTierSelector/CarTierSelector.stories.tsx:3]: Should we call it something like `TierSelector` instead? Just seeing Accordion I thought it was the `Accordion` component we imported :)

File: apps/store/src/components/CarTierSelector/CarTierSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  height: theme.space[7],
+  width: '100%',
+  cursor: 'pointer',
+  display: 'flex',
+  alignItems: 'center',
+  textAlign: 'center',
+  justifyContent: 'space-between',
+  background: theme.colors.gray200,
+  boxShadow: '0px 1px 2px rgba(0, 0, 0, 0.2)',
+  borderRadius: theme.space[3],
```
Comment (gustaveen) [inline apps/store/src/components/CarTierSelector/CarTierSelector.tsx:25]: nit: we can update to the radius values after merge :)

File: apps/store/src/components/CarTierSelector/CarTierSelector.tsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+  borderRadius: '50%',
+  marginRight: '.5rem',
+}))
+
+type RecommendedItemProps = {
+  children: React.ReactNode
+}
+export const RecommendedItem = ({ children }: RecommendedItemProps) => (
+  <RecommendedContainer>
+    <RecommendedDot></RecommendedDot>
```
Comment (gustaveen) [inline apps/store/src/components/CarTierSelector/CarTierSelector.tsx:78]: nit: 
```suggestion
    <RecommendedDot/>
```

**Review by gustaveen — APPROVED** [review]
Great job @KarnellSchultz 🎉

**Comment by gustaveen** [comment]
Come to think of it, will we need tier selector for other products as well? But maybe it's just applicable to car at this point @KarnellSchultz

File: apps/store/src/components/CarTierSelector/CarTierSelector.tsx [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+type HeaderWithTriggerProps = PropsWithChildren<unknown> & {
+  icon?: ReactElement
+}
+const TriggerIcon = styled(ChevronIcon)({
+  transition: 'transform 300ms',
+  '[data-state=open] &': { transform: 'rotate(180deg)' },
+})
+
+export const HeaderWithTrigger = ({
```
Comment (alebedev) [inline apps/store/src/components/CarTierSelector/CarTierSelector.tsx:47]: This seems to duplicate our ui/Accordion.HeaderWithTrigger 
Can we reuse without making it too interconnected?

**Review by alebedev — APPROVED** [review]
Nice!

File: apps/store/src/components/CarTierSelector/CarTierSelector.tsx (hunk 2 above)
Comment (KarnellSchultz) [reply apps/store/src/components/CarTierSelector/CarTierSelector.tsx:25]: Can do

File: apps/store/src/components/CarTierSelector/CarTierSelector.tsx (hunk 4 above)
Comment (KarnellSchultz) [reply apps/store/src/components/CarTierSelector/CarTierSelector.tsx:47]: They have many similarities for sure. This one is based on the version in the `<Accordion />` but imo making these one reusable component would make them unnecessarily complicated and tied to one another.

Having them separate makes it more simple to understand and optimises for future changes.

**Pull request merged**